modules/docs/docgen/plugins/enhancedComponentParser.ts

```diff
@@ -0,0 +1,882 @@
+    node.initializer.expression.expression.expression.text === 'createSubcomponent'
+  ) {
+    const baseElement = getBaseElement(parser, node.initializer.expression.expression.arguments[0]);
+    const options = node.initializer.expression.arguments[0];
+    const modelName = getModelName(parser, options);
+    const subComponents = getSubcomponents(parser, options);
+    const elemPropsHook = getElemPropsHook(parser, options);
+    const displayName = getDisplayName(parser, options);
+    const signature = node.initializer.arguments[0];
+    if (ts.isFunctionLike(signature) && modelName) {
+      const type = node.initializer.typeArguments
+        ? parser.checker.getTypeAtLocation(node.initializer.typeArguments[0])
+        : undefined;
+      const props = type ? getComponentProps(parser, signature, type, baseElement) : [];
+      props.push(getModelProp(parser, modelName));
+      props.push(getElemProp(parser, modelName));
+      const styleComponent = getStyleComponent(displayName, props);
+
+      return {
+        kind: 'enhancedComponent',
+        componentType: 'subcomponent',
+        elemPropsHook,
+        displayName,
+        props: getNonDefaultNonLayoutProps(displayName, props),
+        baseElement,
+        subComponents,
+        styleComponent,
+        model: modelName,
+      };
+    }
+  }
+
+  return undefined;
+});
+
+function getDisplayName(
+  parser: DocParser<SupportedValues>,
+  node: ts.Expression
+): string | undefined {
+  if (t.isObjectLiteralExpression(node)) {
+    const displayProperty = node.properties.find(
+      p => p.name && t.isIdentifier(p.name) && p.name.text === 'displayName'
+    );
+
+    if (
+      displayProperty &&
+      t.isPropertyAssignment(displayProperty) &&
+      t.isStringLiteral(displayProperty.initializer)
+    ) {
+      return displayProperty.initializer.text;
+    }
+  }
+
+  return undefined;
+}
+
+function getNonDefaultNonLayoutProps(
+  displayName: string | undefined,
+  props: ObjectProperty[]
+): ObjectProperty[] {
+  if (displayName && ['Stack', 'Flex', 'Box', 'Grid'].includes(displayName)) {
+    return props;
+  }
+  return props.filter(p => {
+    if (!p.declarations[0]) {
+      // No declaration is found, we'll include this prop
+      return true;
+    }
+    return !p.declarations[0].filePath.includes('layout/lib/utils');
+  });
+}
+
+function getReturnTypeFromElemPropsHook(
+  parser: DocParser<SupportedValues>,
+  signature?: ts.Signature
+): SupportedValues | Value {
+  if (signature) {
+    const type = signature.getReturnType();
+    // A TypeNode is a synthetic AST representation of a type. No truncation removes the `... 12 more ...`
+    const typeNode = parser.checker.typeToTypeNode(
+      type,
+      undefined,
+      ts.NodeBuilderFlags.NoTruncation
+    );
+    if (typeNode) {
+      const value = parser.getValueFromNode(typeNode);
+      if (value.kind === 'object') {
+        value.properties;
+      }
+      // if (value)
```

Comment:
This stood out as a weird comment that might have been code commented out, and maybe needs cleanup?

Reply:
Yep!